src/plugins/intel_cpu/src/nodes/mkldnn_roi_align_node.cpp

```diff
@@ -214,7 +874,16 @@ void MKLDNNROIAlignNode::executeSpecified() {
         }
     }
 
-    for (int n = 0; n < realRois; ++n) {
+    std::vector<int> numSamples(realRois);
+    std::vector<std::shared_ptr<std::vector<float>>> weightsTbl(realRois, 0);
+    std::vector<std::shared_ptr<std::vector<size_t>>> srcAddressListTbl;
+    std::vector<std::shared_ptr<std::vector<int>>> srcIndexTbl;
```

Comment:
Not a call for action, just a thought. Is there any value in precalculation of addresses and indexes in separate pass. Is there any reuse of these computations? Wouldn't it better to merge these computation with the main loop?

Reply:
There are some computed values of ROI reused by each bin and each samples in this ROI, such as relative box(x1,x2,y1,y2), binHeight, binWidth, sampelsX, samplesY. As decide not use nested parallel for now, will have duplicated computation in parallel_3/4d. Anyway, there are potential benefit if only the reused values is precomputed in separate pass and others moved into main loop. I will fine-tune this in a followed-up PR. Thanks Dmitry.